fe/fe-core/src/main/java/com/starrocks/catalog/Function.java

```diff
@@ -374,27 +413,53 @@ private boolean isSubtype(Function other) {
         if (functionName.equalsIgnoreCase("if")) {
             startArgIndex = 1;
         }
-        for (int i = startArgIndex; i < this.argTypes.length; ++i) {
-            // Normally, if type A matches type B, then A and B must be implicitly castable,
-            // but if one of the types is pseudotype, this rule does not hold anymore, so here
-            // we check type matche first.
-            if (other.argTypes[i].matchesType(this.argTypes[i])) {
-                continue;
-            }
-            if (!Type.isImplicitlyCastable(other.argTypes[i], this.argTypes[i], true)) {
+
+        if (other.hasNamedArg()) {
+            if (!this.hasNamedArg() || other.argNames.length != this.argNames.length ||
+                    other.hasVarArgs || this.hasVarArgs) {
                 return false;
             }
-        }
-        // Check trailing varargs.
-        if (this.hasVarArgs) {
-            for (int i = this.argTypes.length; i < other.argTypes.length; ++i) {
-                if (other.argTypes[i].matchesType(getVarArgsType())) {
+            boolean[] mask = new boolean[other.argTypes.length];
+            for (int j = startArgIndex; j < other.argTypes.length; ++j) {
+                boolean found = false;
+                for (int i = startArgIndex; i < this.argTypes.length; ++i) {
+                    if (!mask[i] && this.argNames[i].equals(other.argNames[j])) {
+                        if (!other.argTypes[j].matchesType(this.argTypes[i]) &&
+                                !Type.isImplicitlyCastable(other.argTypes[j], this.argTypes[i], true)) {
+                            return false;
+                        }
+                        found = true;
+                        mask[i] = true;
+                    }
+                }
```

Comment:
I think maybe you need extract some method, like `getArgumentByName` ?

Reply:
cleaned.